pkg/config/derive_test.go

```diff
@@ -0,0 +1,298 @@
+package config
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	acsapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/util/fixtures"
+)
+
+func Test_derived_ImageResourceGroup(t *testing.T) {
+	tests := []struct {
+		want string
+	}{
+		{
+			want: "test-image1",
+		},
+		{
+			want: "test-image2",
+		},
+	}
+	defer os.Setenv("IMAGE_RESOURCEGROUP", os.Getenv("IMAGE_RESOURCEGROUP"))
```

Comment:
this is a bad plan - it alters global state which lays us open to a flake some point down the road.  We need to centralise where we're doing os.Getenv()s to get round this.  That's #466 (@thekad fyi)

Reply:
Short term could consider passing the env var in as a parameter to enable unit testing